DEV-3477: Account Documentation (#785)
cloudposse/docs: docs/layers/accounts/tutorials/how-to-create-and-setup-aws-accounts.mdx

import Steps from '@site/src/components/Steps';
import Step from '@site/src/components/Step';
import StepNumber from '@site/src/components/StepNumber';
import Note from '@site/src/components/Note';
import AtmosWorkflow from '@site/src/components/AtmosWorkflow';

<Intro>
This guide covers the process of creating and setting up new AWS accounts. It includes detailed instructions for updating account catalogs, configuring stack settings, deploying necessary components, and managing AWS account profiles.
</Intro>

## Problem

Setting up new AWS accounts can be complex and error-prone without proper guidance and tooling.

## Solution

This guide provides a step-by-step approach to create and configure new AWS accounts using Cloud Posse's reference architecture components and conventions.

<Steps>
  <Step>
    ### <StepNumber/> Update Account Catalog

    First, add the new account to your account catalog configuration. This defines the account structure within your AWS Organization.

    Navigate to your account catalog file (typically `stacks/catalog/account*.yaml`) and add the new account under the appropriate Organizational Unit (OU).

>     **Example for a new account named `pca`:**

Reviewer: Why pca? Maybe something more example like. 

- acme
- example
- demo
- foobar

    **Deploy Components:**

    ```bash
    # Create the new account
    atmos terraform apply account --stack core-gbl-root

    # Update account map to include the new account
    atmos terraform apply account-map --stack core-gbl-root

    # Configure account settings for the new account
    atmos terraform apply account-settings --stack core-gbl-foobar

    # Deploy AWS Team Roles to enable cross-account access
    atmos terraform apply aws-team-roles --stack core-gbl-foobar
    ```

    <Note title="Component Order">
      The order of deployment is important. The `account` component creates the account, `account-map` updates the account mapping, `account-settings` configures the account, and `aws-team-roles` enables cross-account role assumption.
    </Note>
  </Step>

  <Step>
    ### <StepNumber/> Complete Account Setup via ClickOps

    After deploying the account infrastructure, you need to perform some manual configurations to finalize the account setup.

    **Required Steps:**
    <Steps>
    - Reset the root user password and set up MFA
    - Enable any necessary optional AWS regions
    - Unsubscribe the account's email address from marketing emails

>     </Steps>

Author: let's move that to a follow up task

</Intro>

## Problem

Setting up new AWS accounts can be complex and error-prone without proper guidance and tooling.

## Solution

This guide provides a step-by-step approach to create and configure new AWS accounts using Cloud Posse's reference architecture components and conventions.

<Steps>
  <Step>
    ### <StepNumber/> Update Account Catalog

    First, add the new account to your account catalog configuration. This defines the account structure within your AWS Organization.

    Navigate to your account catalog file (typically `stacks/catalog/account.yaml`) and add the new account under the appropriate Organizational Unit (OU).

    **Example for a new account named `foobar`:**

    ```yaml
    organizational_units:
      - name: core
        accounts:
          # ... existing accounts ...
          - name: foobar
            stage: foobar
            tags:
              eks: false
    ```

    <Note title="Important">
      Choose the appropriate OU based on your organization's account strategy. Common OUs include `core` for core infrastructure and `plat` for platform accounts.
    </Note>
  </Step>

  <Step>
    ### <StepNumber/> Create Stack Configuration Files

>     Create the necessary stack configuration files for your new account using the `stacks/orgs/<namespace>/<tenant>/<stage>` pattern. See the following example where the namespace is `acme`, the tenant is `core`, and the stage is `foobar`.

Reviewer: ```suggestion
    Create the corresponding stack configuration files for your new account in the `stacks/orgs/<namespace>/<tenant>/<stage>` folder (or follow the convention your organization uses). See the following example where the namespace is `acme`, the tenant is `core`, and the stage is `foobar`.
```

    **Example Directory Structure:**

    For a new account `foobar` in the `core` tenant under namespace `acme`:

    ```
    stacks/orgs/acme/core/foobar/
    ├── _defaults.yaml
    ├── global-region/
    │   ├── baseline.yaml          # Account settings
    │   └── identity.yaml          # AWS Team Roles
    └── us-east-1/
        ├── baseline.yaml          # Regional baseline
        └── app.yaml               # Regional application platform
    ```

    <Note title="Recommendation">
      Start with a basic global-region stack for account settings and create placeholder regional stacks for future expansion. You can add more regional stacks and components as your account requirements grow.
    </Note>

    <Steps>
      <Step>

>         ### <StepNumber/> Stage Mixin

Reviewer: ```suggestion
        ### <StepNumber/> Define Stage Mixin
```

> 

Reviewer: ```suggestion
  By convention, we treat every account as an operating stage. Stages are configured as mixins, so that each stack operating in that stage can import that mixin to have a consistent stage name.
```

        **For example:** `stacks/mixins/stage/foobar.yaml`

        Create a mixin file for your stage to keep configuration DRY. This file contains only variables.

        ```yaml
        # This file is for vars only; do not import components here
        # For more information, see "Mixins and Imports with Atmos" in the baseline documentation
        vars:
          stage: foobar
        ```
      </Step>

      <Step>

>         ### <StepNumber/> Account Defaults

Reviewer: ```suggestion
        ### <StepNumber/> Configure Account Defaults
```

        **For example:** `stacks/orgs/acme/core/foobar/_defaults.yaml`

        This file is necessary for keeping configuration DRY and establishing common settings to be imported in all subsequent files.

        ```yaml
        import:
          - orgs/acme/core/_defaults
          - mixins/stage/foobar
        ```
      </Step>

      <Step>

>         ### <StepNumber/> Account Settings

Reviewer: ```suggestion
        ### <StepNumber/> Configure Account Settings
```

        **For example:** `stacks/orgs/acme/core/foobar/global-region/baseline.yaml`

        This file configures account-level settings and policies.

        ```yaml
        import:
          - orgs/acme/core/foobar/_defaults
          - mixins/region/global-region
          - catalog/account-settings

        components:
          terraform:
            account-settings:
              vars:
                # Allow creating public S3 buckets in this account
                # Public buckets are used, for example, to deploy documentation websites into preview environments and serve them via Lambda@Edge
                block_public_acls: false
                ignore_public_acls: false
                block_public_policy: false
                restrict_public_buckets: false
        ```
      </Step>

      <Step>

>         ### <StepNumber/> AWS Team Roles

Reviewer: ```suggestion
        ### <StepNumber/> Configure AWS Team Roles
```

    <Note title="Best Practice">
      Always review configuration changes carefully before merging, as account creation is irreversible and affects your AWS Organization structure.
    </Note>

    Once the PR is reviewed, approved, and merged, proceed to the next step.
  </Step>

  <Step>
    ### <StepNumber/> Deploy Account Infrastructure

    Deploy the necessary components to create and configure your new account. Use `plan` and `apply` commands without `-auto-approve` for safety.

    **Prerequisites:**
    - Ensure you have `SuperAdmin`, `managers`, or appropriate elevated permissions

    **Deploy Components:**

    ```bash
    # Create the new account
    atmos terraform apply account --stack core-gbl-root

    # Update account map to include the new account
    atmos terraform apply account-map --stack core-gbl-root

    # Configure account settings for the new account
    atmos terraform apply account-settings --stack core-gbl-foobar

    # Deploy AWS Team Roles to enable cross-account access
    atmos terraform apply aws-team-roles --stack core-gbl-foobar
    ```

    <Note title="Component Order">

>       The order of deployment is important. The `account` component creates the account, `account-map` updates the account mapping, `account-settings` configures the account, and `aws-team-roles` enables cross-account role assumption.

Reviewer: ```suggestion
      The order of deployment is important. The `account` component creates the account, `account-map` updates the account mapping required to map accounts to IAM roles, `account-settings` configures the account, and `aws-team-roles` enables cross-account role assumption.
```

    </Note>
  </Step>

  <Step>
    ### <StepNumber/> Complete Account Setup via ClickOps

    After deploying the account infrastructure, you need to perform some manual configurations to finalize the account setup.

>     **Required Steps:**

Reviewer: ```suggestion
    **Recommended Steps:**
```

    <Steps>
    - Reset the root user password and set up MFA
    - Enable any necessary optional AWS regions
    - Unsubscribe the account's email address from marketing emails
    </Steps>

>     **Quick Steps:**

Reviewer: This seems very redundant with the recommended steps
